nestkernel/node_collection.h

```diff
@@ -667,6 +678,20 @@ NodeCollection::set_metadata( NodeCollectionMetadataPTR )
   throw KernelException( "Cannot set Metadata on this type of NodeCollection." );
 }
 
+inline size_t
+NodeCollection::get_first() const
+{
+  return ( *begin() ).node_id;
+}
+
+inline size_t
+NodeCollection::get_last() const
+{
+  size_t offset = size() - 1;
+  return ( *( begin() + offset ) ).node_id;
+}
+
+
```

Comment:
I totally realize that it is bad style if
1. one of the original reviewers (me) 
2. comments on a merged pull request
3. by a person not working on NEST anymore

:see_no_evil:

*But*: couldn't this have been done like this?

```suggestion
inline size_t
NodeCollection::get_first() const
{
  return begin()->node_id;
}

inline size_t
NodeCollection::get_last() const
{
  return ( end() - 1 )->node_id;
}
```

Just sayin'...

Reply:
That was not that simple or better saying, things might get more ugly.

The `begin` functions returns an `iterator`, and this iterator is a bit complex, as it hides the distinction between the *primitive* and *composite* types. Thus, `begin()->node_id` won't work and also the `operator-()` is not implemented.